Add notification toast (#1295)
sumup-oss/circuit-ui: .changeset/curly-buttons-compete.md

---
'@sumup/circuit-ui': minor
---

> Added a new `NotificationToast` component that provides quick and contextual feedback based on the outcome of an action.

Reviewer: Not necessarily based on an action, see https://github.com/sumup-oss/circuit-ui/pull/1295#discussion_r766028569 🙂

Reviewer: Maybe we can just align it with the docs

```suggestion
Added a new `NotificationToast` component that provides quick and contextual feedback to the user.
```
